allocate.go

```diff
@@ -524,6 +533,12 @@ type RemoteAllocator struct {
 	wg sync.WaitGroup
 }
 
+type RemoteAllocatorOption struct {
```

Comment:
We should stick with the functional options pattern. Refer to how `ExecAllocatorOption` is implemented.